apix-impl/src/main/java/eu/xenit/apix/alfresco/dictionary/PropertyTypeCheckService.java

```diff
@@ -0,0 +1,69 @@
+package eu.xenit.apix.alfresco.dictionary;
+
+import eu.xenit.apix.data.QName;
+import eu.xenit.apix.properties.PropertyDefinition;
+import eu.xenit.apix.search.nodes.PropertySearchNode;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PropertyTypeCheckService {
+
+    private final static Logger logger = LoggerFactory.getLogger(PropertyTypeCheckService.class);
+
+    private final Map<String, Predicate<String>> dataTypePredicates = new HashMap<String, Predicate<String>>() {{
+        put("d:int", PropertyTypeCheckService::isInt);
+        put("{http://www.alfresco.org/model/dictionary/1.0}int", PropertyTypeCheckService::isInt);
+        put("d:long", PropertyTypeCheckService::isLong);
+        put("{http://www.alfresco.org/model/dictionary/1.0}long", PropertyTypeCheckService::isLong);
+    }};
+
+    private final PropertyService propertyService;
+
+    public PropertyTypeCheckService(PropertyService propertyService) {
+        if (propertyService == null) {
+            logger.warn("Instantiating PropertyTypeCheckService without a PropertyService, type checks will always succeed");
+        }
+        this.propertyService = propertyService;
+    }
+
+    public boolean fitsType(PropertySearchNode node) {
+        if (node == null || node.getValue() == null || node.getName() == null || propertyService == null) {
+            return true;
+        }
+        QName qName = new QName(node.getName().replaceAll("\\\\", "")); //stuff like hyphen in node-dbid is escaped
+        PropertyDefinition propertyDefinition = propertyService.GetPropertyDefinition(qName);
+        if (propertyDefinition == null || propertyDefinition.getDataType() == null) {
+            return true;
+        }
+        for (Entry<String, Predicate<String>> entry : dataTypePredicates.entrySet()) {
```

Comment:
Why do you iterate over the entrySet and compare keyvalues, instead of just calling `dataTypePredicates.get()`?

Reply:
That's better :+1: